test/compress/template-string.js

```diff
@@ -400,3 +400,61 @@ issue_1856_ascii_only: {
     }
     expect_exact: "console.log(`\\\\n\\\\r\\\\u2028\\\\u2029\\n\\r\\u2028\\u2029`);"
 }
+
+side_effects: {
+    options = {
+        evaluate: true,
+        side_effects: true,
+    }
+    input: {
+        `t1`;
+        tag`t2`;
+        `t${3}`;
+        tag`t${4}`;
+        console.log(`
+t${5}`);
+        function f(a) {
+            `t6${a}`;
+            a = `t7${a}` & a;
+            a = `t8${b}` | a;
+            a = f`t9${a}` ^ a;
+        }
+    }
+    expect: {
+        tag`t2`;
+        tag`t${4}`;
+        console.log("\nt5");
+        function f(a) {
+            a &= `t7${a}`;
+            a = `t8${b}` | a;
+            a = f`t9${a}` ^ a;
```

Comment:
Hmm - that's odd it produces `&=` but not `|=` and `^=` in light of:
```js
    var ASSIGN_OPS_COMMUTATIVE = [ '*', '|', '^', '&' ];
```

Reply:
These three tests are stressing `.has_side_effects()` through:

https://github.com/mishoo/UglifyJS2/blob/7d9a8596a9363b4673297a33d4c71ae26f459056/lib/compress.js#L4483-L4490

`` `t8${b}` `` has undeclared variable, whereas ``f`t9${a}` `` effectively contains `f()`

Reply:
Ah - of course! Thanks.